goss-testing/vars/variables-common.yaml

```diff
@@ -66,3 +66,5 @@ k8s_ceph_csi_secrets:
   - csi-sma-secret
 
 kubectl: "/usr/bin/kubectl"
+# false by default--overriden when vshasta is deployed
```

Comment:
I don't see any code which is doing this overriding in this PR.

Reply:
This var went in in a few spots.  It is false by default in the vars files.  There is a hacky override when we run in vshasta to set it to true.  It is false by default so it does nothing on metal installs

Reply:
See https://github.com/Cray-HPE/csm-testing/pull/410/files#diff-fd12da3b417be82600b83d8504e8a7a264a13585ce145769ac363277945447ffR102

Reply:
All of the overrides happen in the vshasta codebase.

Reply:
Why would you not put it in here, if we already have a method for determining that this is a Vshasta system? Seems easier to maintain that way, rather than requiring PRs in two different repos if wanting to make a change to that part of it. No?

Reply:
We certainly could put it in here.  This repo is one of the least developer-friendly repos to contribute to, so that is the main reason I never even considered it.   Also, I already needed to contribute to here, to the csm repo and to csm-rpms in order for my change to show up a few days later, so keeping the vshasta stuff in the vshasta repo meant I could develop it faster. 

I can certainly add it here if that is a gate for you.

Reply:
I would strongly prefer that we have it here. As you've already seen, this repo contains some tests which are not part of any test suite and may or may not still be in use. The same is true of parts of tests and scripts in here -- there are sections which may or may not still be needed. Down the road, if someone notices that we have a lot of vshasta checks, but that there is no place where we ever set it to true, they may conclude that this is another case of something that is no longer in use.

Also, if a change is made to csm-testing that results in the variable files having a different format, or location, or whatever, right now someone making that change would have no way to know that a change would also be required in the vshasta repo. That is another one of the benefits of doing this in the add_local_vars function. It will automatically be kept up to date.

Put another way, I think it's not ideal to have one repo (your vshasta repo) hard-code in assumptions about how another repo (csm-testing) has its stuff internally organized. The automated scripts we provide are really the only interface that should be relied on externally. For example, we don't have anything outside of the repo which directly calls the Goss endpoints, for exactly this reason. There's no promise made that the URLs, port numbers, etc, won't change.

Reply:
Makes sense.  I will make the adjustments

Reply:
For what it's worth, I completely agree with you about this repository not being developer-friendly. That is something we need to improve on.

Reply:
testing after the recent changes

Reply:
It worked

Reply:
The code to set this in the `add_local_vars` function should still be added, using the `is_vshasta_node` function that you created.

Reply:
oops. I'll look

Reply:
ok check now

Reply:
I see the code for the vshasta variable in there, but it looks like it still sets the node manufacturer using ipmitool, which I believe you said will not work for vshasta.

I made this PR against your PR branch:
https://github.com/Cray-HPE/csm-testing/pull/416

It will address that issue, and also make some small formatting changes to make the new code more closely match the style of the existing functions.